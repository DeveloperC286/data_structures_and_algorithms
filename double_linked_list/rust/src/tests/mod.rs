use super::*;

use insta::assert_json_snapshot;

#[test]
fn test_new_double_linked_list() {
    let double_linked_list = DoubleLinkedList::new();
}

#[test]
fn test_add_double_linked_list() {
    // Given
    let mut double_linked_list = DoubleLinkedList::new();
    let node = Node::new(5);

    // When
    double_linked_list.add(node);

    // Then
    assert_json_snapshot!(double_linked_list);
}

#[test]
fn test_multiple_add_double_linked_list() {
    // Given
    let mut double_linked_list = DoubleLinkedList::new();
    let node_1 = Node::new(5);
    let node_2 = Node::new(12);

    // When
    double_linked_list.add(node_1);
    double_linked_list.add(node_2);

    // Then
    assert_json_snapshot!(double_linked_list);
}
