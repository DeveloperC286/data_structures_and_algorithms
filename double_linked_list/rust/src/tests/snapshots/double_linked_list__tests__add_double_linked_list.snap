---
source: src/tests/mod.rs
expression: double_linked_list
---
{
  "length": 1,
  "head": {
    "next": null,
    "previous": null,
    "value": 5
  }
}
